nucypher/network/nodes.py

```diff
@@ -1221,8 +1203,5 @@ def decentralized_identity_evidence(self):
     @property
     def worker_address(self):
         if not self.__worker_address and not self.federated_only:
-            if self.decentralized_identity_evidence is NOT_SIGNED:
-                raise self.StampNotSigned  # TODO: Find a better exception  NRN
-            self.__worker_address = recover_address_eip_191(message=bytes(self.stamp),
-                                                            signature=self.decentralized_identity_evidence)
+            self.__worker_address = to_checksum_address(self.metadata().payload.derive_worker_address())
```

Comment:
This can be called for both local and remote nodes?

Reply:
Yes, they both have the `metadata()` method; for remote ones is what was received from the teacher, for the local one is what it built itself.

Reply:
So is it possible that the derived address conflicts with an address provided by the worker (perhaps my accidental misconfiguration)?  If so how is this handled?

Reply:
Hm, what do you mean? The worker address is not currently a part of metadata.

Reply:
Still thinking through it a bit... I'm wondering if there's a concern of mismatched addresses if the derived address does not match the value of the worker address obtained from user input/config, through `Ursula.__init__`, etc.  Ideally such behavior is not even possible.

Reply:
They way I understand the logic, we initialize an Ethereum account somehow (e.g. from a keystore), and use it to create the evidence. So in order for the derived address not to match, the evidence would have to be derived incorrectly.

Note also that the logic in these lines didn't change - the address was derived from the evidence before as well, it's just that now the evidence is taken from the metadata which takes it from the attribute, instead of directly taking it from the attribute.